src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+     * </p>
+     *
+     * @param src the binary to convert
+     * @param srcPos the position of the lsb to start the conversion
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty, {@code src.length > 8} or
+     *             {@code src.length - srcPos < 4}
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryToHexDigitMsb0_4bits(final boolean[] src, final int srcPos) {
+        if (src.length > 8) {
+            throw new IllegalArgumentException("src.length>8: src.length=" + src.length);
+        }
+        if (src.length - srcPos < 4) {
+            throw new IllegalArgumentException("src.length-srcPos<4: src.length=" + src.length + ", srcPos=" + srcPos);
+        }
+        if (src[srcPos + 3]) {
+            if (src[srcPos + 2]) {
+                if (src[srcPos + 1]) {
+                    return src[srcPos] ? 'f' : '7';
+                }
+                return src[srcPos] ? 'b' : '3';
+            }
+            if (src[srcPos + 1]) {
+                return src[srcPos] ? 'd' : '5';
+            }
+            return src[srcPos] ? '9' : '1';
+        }
+        if (src[srcPos + 2]) {
+            if (src[srcPos + 1]) {
+                return src[srcPos] ? 'e' : '6';
+            }
+            return src[srcPos] ? 'a' : '2';
+        }
+        if (src[srcPos + 1]) {
+            return src[srcPos] ? 'c' : '4';
+        }
+        return src[srcPos] ? '8' : '0';
+    }
+
+    /**
+     * <p>
+     * Converts the first 4 bits of a binary (represented as boolean array) in big endian Msb0
+     * bit ordering to a hexadecimal digit.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) is converted as follow: '8' (1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0) is converted
+     * to '4'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryBeMsb0ToHexDigit(final boolean[] src) {
+        return binaryBeMsb0ToHexDigit(src, 0);
+    }
+
+    /**
+     * <p>
+     * Converts a binary (represented as boolean array) in big endian Msb0 bit ordering to a
+     * hexadecimal digit.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) with srcPos = 0 is converted as follow: '8' (1, 0, 0, 0, 0, 0, 0, 0,
+     * 0, 0, 0, 1, 0, 1, 0, 0) with srcPos = 2 is converted to '5'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @param srcPos the position of the lsb to start the conversion
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryBeMsb0ToHexDigit(boolean[] src, int srcPos) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 31. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 449-460, lines 461-466 into separate methods.